### PR TITLE
Implement SafeMarshal support for hash compare by identity

### PR DESCRIPTION
**Pull request by segiddins**
So there is not a lingering pending test

Intentionally do not support other User Class cases for now

### CODE REVIEW

**Review by martinemde — APPROVED** [review]
Ok, so Marshal considers a Hash set to compare by identity as a user class, but since it's a safe loadable class we specifically allow only that class through the user class loader? Looks correct to me but I'm no Marshal pro.

**Comment by segiddins** [comment]
Yeah, there are even comments in marshal.c about it being a hack

File: lib/rubygems/safe_marshal/visitors/to_ruby.rb [hunk 1]
```diff
@@ -247,6 +247,30 @@ def visit_Gem_SafeMarshal_Elements_Bignum(b)
         end
       end
 
+      def visit_Gem_SafeMarshal_Elements_UserClass(r)
+        if resolve_class(r.name) == ::Hash && r.wrapped_object.is_a?(Elements::Hash)
+
```
Comment (deivid-rodriguez) [inline lib/rubygems/safe_marshal/visitors/to_ruby.rb:252]: Is this blank line intentional?

**Review by deivid-rodriguez — APPROVED** [review]
No Marshal expert either but makes sense to me!

**Comment by deivid-rodriguez** [comment]
I'll merge this since a blank line is no reason to block it :)

**Pull request merged**